### ISSUE TITLE
[nuget add using] nuget packages not added for extension methods.

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#8757]
1. Create a class that uses extension methods from a nuget package. (Below code uses [Extend](https://www.nuget.org/packages/Extend))

``` C#
using System;
using System.Collections.Generic;

namespace ConsoleApplication37
{
    class Program
    {
        static void Main(string[] args)
        {
            var list = new List<String>();
            list.AddIf(x => new Random().Next(0, 10) % 2 == 0, "Value");
            list.AddIfNotContains("Value #1");
            list.AddRange("Value #1", "Value #2", "Value #3");
            list.AddRangeIfNotContains("Value #1", "Value #2", "Value #3");
            list.AddRangeIf(x => new Random().Next(0, 10) % 2 == 0, "Value", "Value #2", "Value #3");
        }
    }
}
```

**Expected**:

``` C#
using System;
using System.Collections.Generic;
using Extend;

namespace ConsoleApplication37
{
    class Program
    {
        static void Main(string[] args)
        {
            var list = new List<String>();
            list.AddIf(x => new Random().Next(0, 10) % 2 == 0, "Value");
            list.AddIfNotContains("Value #1");
            list.AddRange("Value #1", "Value #2", "Value #3");
            list.AddRangeIfNotContains("Value #1", "Value #2", "Value #3");
            list.AddRangeIf(x => new Random().Next(0, 10) % 2 == 0, "Value", "Value #2", "Value #3");
        }
    }
}
```

**Actual**: No fixes are offered

**Comment by jmarolf** [comment]
**NOTE**: If I add the package manually, I am offered to add usings.
![image](https://cloud.githubusercontent.com/assets/9797472/13091685/3807b282-d4b1-11e5-8f3c-d4a89fa1516d.png)

Also, the type it in my NuGetCache.txt
![image](https://cloud.githubusercontent.com/assets/9797472/13091752/7528f8a6-d4b1-11e5-9832-3d507422796d.png)

**Comment by jmarolf** [comment]
@CyrusNajmabadi

**Comment by CyrusNajmabadi** [comment]
Currently the design for Update2.  The issue is that it's non trivial to check applicability properly.  We'd def like to add this in the future.

**Comment by MattGertz** [comment]
Moving to RC due to requirements around .NET Core scenarios.  Might be RC2, though.

**Comment by CyrusNajmabadi** [comment]
Are those requirements specified anywhere?  This is the first i'm hearing about this :)

**Comment by MattGertz** [comment]
@CyrusNajmabadi Still in that meeting; bide a moment & I'll get you the details once I'm out of it.

**Comment by MattGertz** [comment]
This relates to the scenario in [https://github.com/dotnet/roslyn-project-system/issues/583], which is actually an E2E-affecting scenario for the forthcoming .NET Core release (scenario S4, first user story -- I will send you the scenario link offline).  @morrisjoe or @balachir can give you more details on the scenario.  Not fixing this is apparently a regression from previous .NET Core usage.

**Comment by CyrusNajmabadi** [comment]
tagging @srivatsn   Sri, if we wanted to add extension-method information to the Nuget index, how would we go about doing that?   I'm curious what the size impact would be as well.  Thanks!

**Comment by balachir** [comment]
Correct, this is a regression from Dev14 tooling behavior for ASP.NET Core projects. See the EXPECTED in dotnet/roslyn-project-system#583 - that's what you got with Dev14. 

@abpiskunov did the implementation for this in Dev14. You could chat with him about ideas if needed. 

Couple of examples with empty ASP.NET Core web app:
1. Type `services.AddMvc` in ConfigureServices method of Startup.cs and hit Ctrl+.
2. Type `app.UseStaticFiles` in Configure method of Startup.cs and hit Ctrl+.

**Comment by CyrusNajmabadi** [comment]
I think making this work is not an issue.  The primary concern would be about what this would do to the size of the package index we download, and how much more ram it would need to keep that index in memory.

We'll need Sri to weigh in on the feasibility of that, and if it adds an unacceptable amount of cost to this scenario.  This will also affect things like how much cost we have for downloading this index as well.

**Comment by Pilchie** [comment]
Moving to RC.3 since the index does not yet contain extension methods.

**Comment by Pilchie** [comment]
Also reported in https://developercommunity.visualstudio.com/content/problem/29937/quick-actions-ctrl-does-not-provide-option-to-inst.html.

**Comment by JVaghela-Fintech** [comment]
Any update on this??

**Comment by CyrusNajmabadi** [comment]
@JVaghela-Fintech there is no update on this.

**Comment by jnm2** [comment]
I think about this whenever I start a new test project because I use Shouldly for assertions.
